libraries/botbuilder-dialogs/src/dialogManager.ts

```diff
@@ -51,64 +53,74 @@ export interface DialogManagerConfiguration {
 }
 
 export class DialogManager extends Configurable {
-    private dialogSet: DialogSet = new DialogSet();
-    private rootDialogId: string;
-    private dialogStateProperty: string;
+    private _rootDialogId: string;
+    private readonly _dialogStateProperty: string;
     private readonly _initialTurnState: TurnContextStateCollection = new TurnContextStateCollection();
 
     public constructor(rootDialog?: Dialog, dialogStateProperty?: string) {
         super();
         if (rootDialog) { this.rootDialog = rootDialog; }
-        this.dialogStateProperty = dialogStateProperty || 'DialogStateProperty';
+        this._dialogStateProperty = dialogStateProperty || 'DialogStateProperty';
         this._initialTurnState.set(DialogTurnStateConstants.dialogManager, this);
     }
 
+    /**
+     * Bots persisted conversation state.
+     */
+    public conversationState: ConversationState;
+
+    /**
+     * Optional. Bots persisted user state.
+     */
+    public userState?: UserState;
+
     /**
      * Values that will be copied to the `TurnContext.turnState` at the beginning of each turn.
      */
     public get initialTurnState(): TurnContextStateCollection {
         return this._initialTurnState;
     }
 
-    /**
-     * Bots persisted conversation state.
-     */
-    public conversationState: ConversationState;
-
     /**
      * Root dialog to start from [onTurn()](#onturn) method.
      */
-    public set rootDialog(dialog: Dialog) {
-        this.dialogSet.add(dialog);
-        this.rootDialogId = dialog.id;
+    public set rootDialog(value: Dialog) {
+        this.dialogs = new DialogSet();
+        if (value) {
+            this._rootDialogId = value.id;
+            this.dialogs.telemetryClient = value.telemetryClient;
+            this.dialogs.add(value);
+        } else {
+            this._rootDialogId = undefined;
+        }
```

Comment:
This is a bit more complicated than the original logic... Can you add tests for it?

Reply:
Tests added, this is ported from c# version of `rootDialog`'s set method.